src/pip/_internal/cli/req_command.py

```diff
@@ -349,3 +353,6 @@ def _build_package_finder(
             selection_prefs=selection_prefs,
             target_python=target_python,
         )
+
+
+_CommandT = TypeVar("_CommandT", bound=RequirementCommand)
```

Comment:
Aren't TypeVars usually defined shortly before they're first used or shortly after the imports or even in the `if TYPE_CHECKING` block?

Regardless, please move it somewhere other than the very end of the module.

Reply:
IIRC mypy will tolerate a forward reference in a typevar bound just fine.

```python
from typing import TypeVar

T = TypeVar("T", bound="Base")

def do_something(obj: T) -> T:
	return obj

class Base:
	pass

class Wrong:
	pass

do_something(Base())   # passes
do_something(Wrong())  # E: Value of type variable "T" of "do_something" cannot be "Wrong"  [type-var]
```